src/plugins/intel_cpu/src/nodes/pad.cpp

```diff
@@ -157,112 +187,169 @@ void Pad::initSupportedPrimitiveDescriptors() {
     auto canUseBlocked = [&](const size_t blockSize) {
         const auto& srcDims = inputDataShape.getDims();
         return srcDims[1] != Shape::UNDEFINED_DIM && srcDims[1] % blockSize == 0 &&
-               ((attrs.padMode == CONSTANT && attrs.padsBegin[1] % blockSize == 0 && attrs.padsEnd[1] % blockSize == 0) ||
-               (attrs.padMode != CONSTANT && attrs.padsBegin[1] == 0 && attrs.padsEnd[1] == 0));
+               ((attrs.padMode == CONSTANT && attrs.padsBegin[1] % blockSize == 0 &&
+                 attrs.padsEnd[1] % blockSize == 0) ||
+                (attrs.padMode != CONSTANT && attrs.padsBegin[1] == 0 && attrs.padsEnd[1] == 0));
     };
 
     if (numOfDims == 4 || numOfDims == 5) {
-        if (canUseBlocked(8))
-            pushSupportedPrimitiveDescriptor(LayoutType::nCsp8c);
-        if (canUseBlocked(16))
-            pushSupportedPrimitiveDescriptor(LayoutType::nCsp16c);
+        if (!shapeHasDataDependency) {
+            if (canUseBlocked(8))
+                pushSupportedPrimitiveDescriptor(LayoutType::nCsp8c);
+            if (canUseBlocked(16))
+                pushSupportedPrimitiveDescriptor(LayoutType::nCsp16c);
+        }
     }
 }
 
+bool Pad::needShapeInfer() const {
+    return Node::inputShapesModified() || shapeHasDataDependency;
+}
+
 void Pad::createPrimitive() {
-    auto& dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
-    auto& srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
+    if (inputShapesDefined() && isExecutable() && !shapeHasDataDependency) {
+        prepareParams();
```

Comment:
Probably I miss something... But seems like we should override method [needPrepareParams()](https://github.com/openvinotoolkit/openvino/blob/e409f708e03cb7b88d2d6c2b732395d30a1fcb61/src/plugins/intel_cpu/src/node.cpp#L1466) because base method returns `true` on only cases when input shapes are changed. But in this case it isn't enough, as I see. If `Pad` node has the same input shapes with different paddings on each inference request, we won't recalculate new attributes but we have to do it.
So I'd suggest to override [needPrepareParams()](https://github.com/openvinotoolkit/openvino/blob/e409f708e03cb7b88d2d6c2b732395d30a1fcb61/src/plugins/intel_cpu/src/node.cpp#L1466), maybe using the following code part:
```
bool Pad::needPrepareParams() const {
    return inputShapesModified() || shapeHasDataDependency;
}
```

Could you please take a look on this case?